### PR TITLE
Request parser

### LINKED ISSUE #10: Http parser

### PR DESCRIPTION
**Pull request by maartenRamadan**

### CODE REVIEW

File: src/server/http/HTTPClientData.cpp [hunk 1]
```diff
@@ -3,21 +3,151 @@
 //
 
 #include "server/http/HTTPClientData.hpp"
+#include <iostream>
+
```
Comment (pmerku) [inline src/server/http/HTTPClientData.cpp:7]: Adding `#include <vector>` should be enough to fix the compilation

File: src/server/http/ft_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		char c;
+		size_t i = 0;
+		size_t ret = 0;
+		size_t abc = 10;
+		
+		while (str[i] == ' ')
+			i++;
+		while (i < str.length())
+		{
+			if (std::isdigit(str[i]))
```
Comment (pmerku) [inline src/server/http/ft_utils.cpp]: I don't know if we can use `isdigit()`

File: src/server/http/ft_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+				ret = ret * 16 + (str[i] - '0');
+			else {
+				c = std::toupper(str[i]);
```
Comment (pmerku) [inline src/server/http/ft_utils.cpp]: And this too

**Review by pmerku — CHANGES_REQUESTED** [review]

**Review by mrjvs — CHANGES_REQUESTED** [review]
talked about in meeting

**Pull request closed without merging**